examples/hpc/dla-driver/src/lib.rs

```diff
@@ -245,65 +272,74 @@ impl Dla {
         res
     }
 
+    /// Reads len amount of bytes from DLA's output bank(s)
     pub fn read_output(&self, len: usize) -> Vec<i8> {
         // VP only support reading from banks
         if cfg!(feature = "vp") {
-            return self.read_data_bank(&self.get_output_bank(), len);
+            return self.read_data_bank(self.get_output_bank(), len);
         }
-        self.read_data_bank(&MemoryBank::Bank0, len)
+        self.read_data_bank(MemoryBank::Bank0, len)
     }
 
+    /// Reads len amount of bytes from DLA's input bank(s)
     pub fn read_input_bank(&self, len: usize) -> Vec<i8> {
-        self.read_data_bank(&self.get_input_bank(), len)
+        self.read_data_bank(self.get_input_bank(), len)
     }
 
+    /// Reads len amount of bytes from DLA's weight bank(s)
     pub fn read_weight_bank(&self, len: usize) -> Vec<i8> {
-        self.read_data_bank(&self.get_kernel_bank(), len)
+        self.read_data_bank(self.get_kernel_bank(), len)
     }
 
+    /// Writes buffer to DLA's input bank(s)
     pub fn write_input(&self, input: &mut [i8]) {
         // TODO optimize memory bank logic
-        self.write_data_bank(self.get_input_bank().addr(), input)
+        self.write_data_bank(self.get_input_bank().offset(), input)
     }
 
+    /// Writes buffer to DLA's kernel bank(s)
     pub fn write_kernel(&self, kernel: &mut [i8]) {
         // TODO optimize memory bank logic
-        self.write_data_bank(self.get_kernel_bank().addr(), kernel)
+        self.write_data_bank(self.get_kernel_bank().offset(), kernel)
     }
 
+    /// Sets one of the DLA's memory banks as starting bank for inputs
     fn set_input_data_bank(&self, bank: MemoryBank) {
         let mut reg = self.read_u32(DLA_BUF_DATA_BANK);
         reg = set_bits!(
             DLA_BUF_DATA_BANK_B_OFFSET,
             DLA_BUF_DATA_BANK_B_BITMASK,
             reg,
-            bank.value()
+            bank.from()
```

Comment:
Yeah this looks wrong :D see above. This should definitely be an `into`.

Reply:
changed to into